joint_calling/utils.py

```diff
@@ -1,32 +1,175 @@
 """Utility functions for the cpg-qc module"""
 
 import os
+import subprocess
+import tempfile
+import logging
 import sys
 import time
 import hashlib
-from os.path import isdir, isfile, exists
-from typing import Any, Callable
+from os.path import isdir, isfile, exists, join, basename
+from typing import Any, Callable, List
+import shutil
 
+import pandas as pd
 import hail as hl
 import click
 from google.cloud import storage
+from joint_calling import _version
+
+
+logger = logging.getLogger('joint-calling')
+logger.setLevel('INFO')
 
 
 DEFAULT_REF = 'GRCh38'
 
 
-def init_hail(name: str, local_tmp_dir: str):
+DATAPROC_PACKAGES = [
+    'joint-calling',
+    'click',
+    'cpg-gnomad',
+    'google',
+    'slackclient',
+    'fsspec',
+    'sklearn',
+    'gcloud',
+]
+
+
+GATK_VERSION = '4.2.0.0'
+GATK_DOCKER = f'us.gcr.io/broad-gatk/gatk:{GATK_VERSION}'
+# GnarlyGenotyper is in Beta and crashes with NullPointerException when using the
+# official GATK docker, that's why using a separate image for it:
+GNARLY_DOCKER = 'gcr.io/broad-dsde-methods/gnarly_genotyper:hail_ukbb_300K'
```

Comment:
Probably worth copying these images over to our AR?

```suggestion
GATK_DOCKER = f'us.gcr.io/broad-gatk/gatk:{GATK_VERSION}'
# GnarlyGenotyper is in Beta and crashes with NullPointerException when using the
# official GATK docker, that's why we're using a separate image for it:
GNARLY_DOCKER = 'gcr.io/broad-dsde-methods/gnarly_genotyper:hail_ukbb_300K'
```

Reply:
Moved the containers to the cpg-common AR.
https://console.cloud.google.com/artifacts/docker/cpg-common/australia-southeast1/joint-calling?project=cpg-common
https://github.com/populationgenomics/joint-calling/blob/add-rf-and-vqsr/docker/copy-images.sh